lib/tools/system-calls.js

```diff
@@ -779,31 +795,19 @@ systemCallMethods.waitForDevice = async function (appDeviceReadyTimeout = 30) {
  */
 systemCallMethods.reboot = async function (retries = DEFAULT_ADB_REBOOT_RETRIES) {
   try {
-    try {
-      await this.shell(['stop']);
-    } catch (err) {
-      if (err.message.indexOf('must be root') === -1) {
-        throw err;
-      }
-      // this device needs adb to be running as root to stop.
-      // so try to restart the daemon
-      log.debug('Device requires adb to be running as root in order to reboot. Restarting daemon');
-      await this.root();
-      await this.shell(['stop']);
-    }
-    await B.delay(2000); // let the emu finish stopping;
+    await this.privilegedShell(['stop']);
+    await B.delay(EMU_STOP_TIMEOUT); // let the emu finish stopping;
     await this.setDeviceProperty('sys.boot_completed', 0);
-    await this.shell(['start']);
+    await this.privilegedShell(['start']);
     await retryInterval(retries, 1000, async () => {
       let booted = await this.getDeviceProperty('sys.boot_completed');
       if (booted === '1') {
         return;
-      } else {
-        // we don't want the stack trace, so no log.errorAndThrow
-        let msg = 'Waiting for reboot. This takes time';
-        log.debug(msg);
-        throw new Error(msg);
       }
+      // we don't want the stack trace, so no log.errorAndThrow
+      let msg = 'Waiting for reboot. This takes time';
+      log.debug(msg);
+      throw new Error(msg);
     });
   } finally {
     await this.unroot();
```

Comment:
this would probably be not needed anymore if unroot is done inside the helper method